engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorContextProperties.java

```diff
@@ -0,0 +1,173 @@
+
+	/**
+	 * The default traversable resolver for this factory.
+	 */
+	private final TraversableResolver traversableResolver;
+
+	/**
+	 * The default parameter name provider for this factory.
+	 */
+	private final ExecutableParameterNameProvider parameterNameProvider;
+
+	/**
+	 * Provider for the current time when validating {@code @Future} or {@code @Past}
+	 */
+	private final ClockProvider clockProvider;
+
+	/**
+	 * Defines the temporal validation tolerance i.e. the allowed margin of error when comparing date/time in temporal
+	 * constraints.
+	 */
+	private final Duration temporalValidationTolerance;
+
+	/**
+	 * Used to get the {@code ScriptEvaluatorFactory} when validating {@code @ScriptAssert} and
+	 * {@code @ParameterScriptAssert} constraints.
+	 */
+	private final ScriptEvaluatorFactory scriptEvaluatorFactory;
+
+
+	/**
+	 * Hibernate Validator specific flag to abort validation on first constraint violation.
+	 */
+	private final boolean failFast;
+
+	/**
+	 * Hibernate Validator specific flag to disable the {@code TraversableResolver} result cache.
+	 */
+	private final boolean traversableResolverResultCacheEnabled;
+
+	private ValidatorContextProperties(MessageInterpolator messageInterpolator, TraversableResolver traversableResolver, ExecutableParameterNameProvider parameterNameProvider, ClockProvider clockProvider, Duration temporalValidationTolerance, ScriptEvaluatorFactory scriptEvaluatorFactory, boolean failFast, boolean traversableResolverResultCacheEnabled) {
+		this.messageInterpolator = messageInterpolator;
+		this.traversableResolver = traversableResolver;
+		this.parameterNameProvider = parameterNameProvider;
+		this.clockProvider = clockProvider;
+		this.temporalValidationTolerance = temporalValidationTolerance;
+		this.scriptEvaluatorFactory = scriptEvaluatorFactory;
+		this.failFast = failFast;
+		this.traversableResolverResultCacheEnabled = traversableResolverResultCacheEnabled;
+	}
+
+	public MessageInterpolator getMessageInterpolator() {
+		return this.messageInterpolator;
+	}
+
+	public TraversableResolver getTraversableResolver() {
+		return this.traversableResolver;
+	}
+
+	public ExecutableParameterNameProvider getParameterNameProvider() {
+		return this.parameterNameProvider;
+	}
+
+	public ClockProvider getClockProvider() {
+		return this.clockProvider;
+	}
+
+	public Duration getTemporalValidationTolerance() {
+		return this.temporalValidationTolerance;
+	}
+
+	public ScriptEvaluatorFactory getScriptEvaluatorFactory() {
+		return this.scriptEvaluatorFactory;
+	}
+
+	public boolean isFailFast() {
+		return this.failFast;
+	}
+
+	public boolean isTraversableResolverResultCacheEnabled() {
+		return this.traversableResolverResultCacheEnabled;
+	}
+
+	public TraversableResolver getWrapedWithCachingForSingleValidation() {
+		return TraversableResolvers.wrapWithCachingForSingleValidation( traversableResolver, traversableResolverResultCacheEnabled );
+	}
+
+	public static class Builder {
```

Comment:
Wondering if this builder pattern is bringing something to the plate. Looks like it's just noise. Moreover it might be error prone as we might forget to set up one of the necessary elements.

Reply:
well, I just don't like constructors with a huge amount of parameters :), but other than that, I think  there's not much benefit from it. Let me know if I should just remove it.

Reply:
I tend to agree but I think it would be an error in this case. We instantiate the context in several places and with the builder, we could forget one of the arguments.

I think in this case, a multi-parameters constructor is best.